test/chai-sql.coffee

```diff
@@ -101,12 +103,21 @@ exports.operandToAbstractSQLFactory = (binds = [], defaultResource = 'pilot', de
 			fieldParts = operand.split('/')
 			if fieldParts.length > 1
 				alias = parentAlias
-				for fieldPart in fieldParts[...-2]
-					alias = "#{alias}.#{fieldPart.replace(/__/g, '-')}"
-				mapping = clientModel.resourceToSQLMappings[fieldParts[fieldParts.length - 2]][fieldParts[fieldParts.length - 1]]
-				mapping = ["#{alias}.#{mapping[0]}", mapping[1...]...]
+				previousResource = _(parentAlias).split('.').last()
+				for resourceName in fieldParts[...-1]
+					sqlName = resourceNameToSqlName(resourceName)
+					sqlNameParts = sqlName.split('-')
+					mapping = _.get(clientModel.relationships[previousResource], sqlNameParts.join('.'))
+					refTable = mapping[1][0]
+					if sqlNameParts.length > 1 and not _.includes(refTable, '-')
+						# Add the verb to tables that don't include the verb already
+						alias = "#{alias}.#{sqlNameParts[0]}-#{refTable}"
+					else
+						alias = "#{alias}.#{refTable}"
+					previousResource = refTable
+				mapping = [alias, _.last(fieldParts)]
```

Comment:
Oh, I see. I was confused because I took "for blah in fieldParts[...-1]" to mean "iterate over the elements of the last element of fieldParts". My bad.